src/Radicle/Internal/PrimFns.hs

```diff
@@ -339,13 +377,56 @@ purePrimFns = PrimFns $ fromList $ first Ident <$>
 -- * Helpers
 
 -- Many primFns have a single argument.
-oneArg :: Monad m => Text -> (Value -> Lang m Value) -> [Value] -> Lang m Value
-oneArg fname f = \case
-  [x] -> f x
-  xs -> throwErrorHere $ WrongNumberOfArgs fname 1 (length xs)
+oneArg :: Monad m =>
+          Text
+       -> D.Pan
+       -> Text
+       -> D.Pan
+       -> D.Value
+       -> D.Pan
+       -> D.Value
+       -> (Value -> Lang m Value)
+       -> (Ident, Maybe (D.Described D.Value), [Value] -> Lang m Value)
```

Comment:
This isn't the most wonderful function in the world. This arg stuff was already begging for reconsideration, but now I think it's been pushed over some edge. There are a few improvements to consider:

- a) Simplest: add a record datatype for one/two/threeArgs
- b) Add a single record type with list of args, no check that docs match number of args in fn
- c) Like (b), but use an hlist to make sure numbers do match
- d) Add singletons for prisms into `Value`, and make those singletons instance of a class that gives them types. Possibly make a class to recurse on number of args. Make sure number of arg documentation matches number of args.

Reply:
What option do you prefer?